mtp_bank_admin/apps/bank_admin/statement.py

```diff
@@ -0,0 +1,117 @@
+import time
+import datetime
+
+from django.conf import settings
+from bai2 import bai2, models, constants
+from moj_auth import api_client
+
+from . import BAI2_STMT_LABEL
+from .exceptions import EmptyFileError
+from .utils import retrieve_all_transactions, get_transaction_uid,\
+    create_batch_record
+
+SENDER_ID = 'NWBKGB2L'
+RECEIVER_ID = '391796'
+ACCOUNT_NUMBER = '10002383 607080'
+CURRENCY = 'GBP'
+RECORD_LENGTH = 80
+
+
+def generate_bank_statement(request):
+    transactions = retrieve_all_transactions(request, 'credited,refunded',
+                                             exclude_batch_label=BAI2_STMT_LABEL)
+
+    if len(transactions) == 0:
+        raise EmptyFileError()
+
+    transaction_records = []
+    credit_num = 0
+    credit_total = 0
+    debit_num = 0
+    debit_total = 0
+    for transaction in transactions:
+        transaction_record = models.TransactionDetail([])
+        if transaction.get('refunded', False):
+            transaction_record.type_code = constants.TypeCodes['699']
+            debit_num += 1
+            debit_total += transaction['amount']
+        else:
+            transaction_record.type_code = constants.TypeCodes['399']
+            credit_num += 1
+            credit_total += transaction['amount']
+
+        transaction_record.amount = transaction['amount']
+        transaction_record.text = str(get_transaction_uid(transaction))
+        transaction_records.append(transaction_record)
+
+    bai2_file = models.Bai2File()
+
+    file_header = bai2_file.header
+    file_header.sender_id = SENDER_ID
+    file_header.receiver_id = RECEIVER_ID
+    file_header.creation_date = datetime.date.today()
+    file_header.creation_time = datetime.datetime.utcnow().time()
+    file_header.file_id = int(time.time()) % 86400
```

Comment:
I assume the `86400` is 24 hours in seconds? Is `file_id` just “seconds since the start of today”? If so: why, and could we make that clearer?